countries/src/main/java/com/lambdaschool/countries/AgeController.java

```diff
@@ -0,0 +1,62 @@
+package com.lambdaschool.countries;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+
+@RestController
+@RequestMapping("/age")
+public class AgeController
+{
+    // localhost:2019/age/age/45
+    @GetMapping(value = "/age/{age}",
+            produces = {"application/json"})
+    public ResponseEntity<?> findOlderPeople(
+            @PathVariable
+                    long age)
+    {
+        ArrayList<Country> rtnCountries = CountriesApplication.countryList.
+                findCountries(c -> c.getMedianAge() >= age);
+
+        return new ResponseEntity<>(rtnCountries, HttpStatus.OK);
+    }
+
+    // localhost:2019/age/min
+    @GetMapping(value = "/min",
+            produces = {"application/json"})
+    public ResponseEntity<?> findMinAge()
+    {
+        CountriesApplication.countryList.countryList.sort((c1, c2) -> (c1.getMedianAge() - c2.getMedianAge()));
+
+        Country rtnCountry = CountriesApplication.countryList.countryList.get(0);
+        return new ResponseEntity<>(rtnCountry, HttpStatus.OK);
+    }
+
+    // localhost:2019/age/max
+    @GetMapping(value = "/max",
+            produces = {"application/json"})
+    public ResponseEntity<?> findMaxAge()
+    {
+        CountriesApplication.countryList.countryList.sort((c1, c2) -> (c2.getMedianAge() - c1.getMedianAge()));
+
+        Country rtnCountry = CountriesApplication.countryList.countryList.get(0);
+        return new ResponseEntity<>(rtnCountry, HttpStatus.OK);
+    }
+
+    // localhost:2019/age/median
+    @GetMapping(value = "/median",
+            produces = {"application/json"})
+    public ResponseEntity<?> findMedianAge()
+    {
+        CountriesApplication.countryList.countryList.sort((c1, c2) -> (int) (c1.getMedianAge() - c2.getMedianAge()));
+
+        Country rtnCountry = CountriesApplication.countryList.countryList.get((CountriesApplication.countryList.countryList.size() / 2) + 1);
+        return new ResponseEntity<>(rtnCountry, HttpStatus.OK);
+    }
+
```

Comment:
Minor issue - in computing for the median, don't throw in a `+ 1` because that will actually take you to the index right after the median's.

For example: length of 5 / 2 -> 2, which would be the middle in `0 1   2   3 4` -> adding one pushes you out of the middle.

Another example with even length: 6 / 2 -> 3, which would be among the middle two in `0 1   2 3   4 5` -> adding one pushes you out of the middle too.